flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -428,6 +429,13 @@ public ProgramTargetDescriptor executeUpdate(String sessionId, String statement)
         return executeUpdateInternal(sessionId, context, statement);
     }
 
+    @VisibleForTesting
+    List<ModuleEntry> listFullModules(String sessionId) throws SqlExecutionException {
```

Comment:
I think we will need this interface in the next `SHOW FULL MODULES` PR, we can add this method into base interface and implement it beside `listModules`.

Reply:
Hi Jark, I've also considered this point before issuing this PR. 

I thought the reason we have this method `LocalExecutor#listModules()` is because the `SHOW MODULES` syntax is not supported by the `FlinkSqlParserImpl`and `TableEnvironment` at that moment. Thus we cannot perform `localExecutor.executeSql("SHOW MODULES")`. 

Once the `SHOW [FULL] MODULES` syntax is supported, all `SHOW MODULES` commands would go for the table result (except for `SHOW MODULES` using the old planner). That's why I didn't make `listFullModules()` an interface method. I was meant to make it a testing method.

Towards the next PR, since `SHOW FULL MODULES` will display the module's use status, `TableSchema` is needed for the result and print as tableau form.  The extra effort will be put in if calling `localExecutor.listFullModules()` directly. However, we can reuse `PrintUtils#printAsTableauForm` if using `localExecutor.executeSql("SHOW FULL MODULES")`.

Please correct me if  I miss something or not understand wrong:-)

Reply:
Thanks for the explanation.